### PR TITLE
Investigate background pop overwritting

### PR DESCRIPTION
**Pull request by michael-harper**
In attempting to remove related individuals from supplied background populations (`hgdp` and `thousand-genomes`) there is a bug where only the last supplied dataset is having related individuals removed. 

The fix is to indent the subset of the `dense_mt` to within the for-loop where we are removing related individuals from background populations.

Relevant Slack threads here:
- https://centrepopgen.slack.com/archives/C03FA2M1MR9/p1722387137408999
- https://centrepopgen.slack.com/archives/C03FA2M1MR9/p1722893919945029 (refers to Dataproc issue that cropped up while trying to fix this issue)

### CODE REVIEW

File: cpg_workflows/large_cohort/ancestry_pca.py [hunk 1]
```diff
@@ -86,17 +90,17 @@ def add_background(
                 )
             else:
                 logging.info('No related samples to drop from background dataset')
+
+            # save metadata info before merging dense and background datasets
+            ht = background_mt.cols()
+            background_mt = background_mt.select_cols().select_rows().select_entries('GT', 'GQ', 'DP', 'AD')
+            background_mt = background_mt.naive_coalesce(5000)
+            # combine dense dataset with background population dataset
+            dense_mt = dense_mt.union_cols(background_mt)
```
[the remark this answers is not in the document]
Comment (michael-harper) [reply cpg_workflows/large_cohort/ancestry_pca.py:99]: Reading the [documentation](https://hail.is/docs/0.2/hail.MatrixTable.html#hail.MatrixTable.select_cols) it looks like `select_cols()` with no parameters just selects all columns:
> Select existing column fields or create new fields by name, dropping the rest.

Which seems a bit redundant unless this line was to make it obvious we're taking all columns and rows 

I tested in a Jupyter notebook and it seems to keep all columns if left blank
Screenshot 2024-07-30 at 9 28 58 AM

[the remark this answers is not in the document]
Comment (KatalinaBobowik) [reply cpg_workflows/large_cohort/ancestry_pca.py:100]: `select_cols()` will only select the column fields that the mt is keyed by, unless specified within the function

**Comment by KatalinaBobowik** [comment]
Has this worked in `test`?

**Comment by michael-harper** [comment]
@KatalinaBobowik yep - here's the [Slack thread](https://centrepopgen.slack.com/archives/C03FA2M1MR9/p1722893919945029)

**Review by KatalinaBobowik — APPROVED** [review]

**Pull request merged**